examples/src/scala/org/pantsbuild/example/strict_deps/BUILD

```diff
@@ -0,0 +1,32 @@
+# Copyright 2014 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+scala_library(
+  name = "a",
+  sources = ["A.scala"],
+)
+
+scala_library(
+  name = "b",
+  sources = ["B.scala"],
+)
+
+scala_library(
+  name = "c",
+  sources = ["C.scala"],
+  dependencies = [":a"],
+)
+
+scala_library(
+  name = "d",
+  sources = ["D.scala"],
+  dependencies = [":b", ":c"],
+  strict_deps = True,
+)
+
+scala_library(
+  name = "e",
+  sources = ["E.scala"],
+  dependencies = [":b", ":c"],
+  strict_deps = False,
+)
```

Comment:
Maybe worth nothing somewhere that an issue here:
1. export-dep-as-jar `d` looks good:
```
        "examples/src/scala/org/pantsbuild/example/strict_deps:d": {
            "targets": [],
            "libraries": [
                "examples.src.scala.org.pantsbuild.example.strict_deps.b",
                "org.scala-lang:scala-library:2.12.8",
                "examples.src.scala.org.pantsbuild.example.strict_deps.c"
            ],
```
where b and c are treated as libraries

2. export-dep-as-jar d & c will be slightly problematic:
```
        "examples/src/scala/org/pantsbuild/example/strict_deps:d": {
            "targets": [
                "examples/src/scala/org/pantsbuild/example/strict_deps:c"
            ],
            "libraries": [
                "examples.src.scala.org.pantsbuild.example.strict_deps.b",
                "org.scala-lang:scala-library:2.12.8"
            ],
...
        "examples/src/scala/org/pantsbuild/example/strict_deps:c": {
            "targets": [],
            "libraries": [
                "examples.src.scala.org.pantsbuild.example.strict_deps.a",
                "org.scala-lang:scala-library:2.12.8"
            ],
```
Because `d` and `c` here are modules in IntelliJ, but IntelliJ doesn't understand strict deps, so library `a` will be accessible from module `d` here. That said, this may not be a huge issue.

Reply:
I see. That's probably what #9146 is for.

Reply:
Yep :) I wanted to make them distinct PRs, because they solve different problems (this one is a bugfix, the other one is a semantic change).